week-2/Homework/mandatory/3-playing-computer.js

```diff
@@ -7,12 +7,13 @@
   Answer the following questions:
 
   1. This program throws an error. Why? (If you can't find it, try executing it).
+   //answer :console.log(b);
   2. Remove the line that throws the error.
-  3. What is printed to the console?
-  4. How many times is "f1" called?
-  5. How many times is "f2" called?
-  6. What value does the "a" parameter take in the first "f1" call?
-  7. What is the value of the "a" outer variable when "f1" is called for the first time?
+  3. What is printed to the console?//answer: b is not defined
+  4. How many times is "f1" called? //answer : 3 times
```

Comment:
Unfortunately 4-7 are all slightly off - can you give them another go? It may help to get some paper and write down each of the loop iterations, filling in the ?s for each iteration:
```
i = 0: a = ?, i % 2 = ?, d = f2(?, ?) = ?, e = f1(?, ?) = ?, console.log(?)
i = 1: a = ?, ...
```